### PR TITLE
Enable log retention for cloudwatch log groups

### PR DESCRIPTION
**Pull request by yuriipolishchuk**
# PR o'clock

## Description

Enable log retention for cloudwatch log groups. 
https://www.terraform.io/docs/providers/aws/r/eks_cluster.html#enabling-control-plane-logging

Breaking change:
If logging is already enabled, the cloudwatch log group resource cannot be created.
As a workaround disable logging, delete old log group and apply plan again.

### Checklist

- [x] `terraform fmt` and `terraform validate` both work from the root and `examples/*` directories
- [x] CI tests are passing
- [x] I've added my change to CHANGELOG.md and highlighted any breaking changes
- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

File: version [hunk 1]
```diff
@@ -1 +1 @@
-v4.0.2
+v4.0.3
```
Comment (max-rocket-internet) [inline version]: Please leave this file as is, it's only changed when a new release is created.

Comment (yuriipolishchuk-finsight) [reply version]: @max-rocket-internet done

Comment (yuriipolishchuk) [reply version]: @max-rocket-internet done

File: variables.tf [hunk 2]
```diff
@@ -4,6 +4,11 @@ variable "cluster_enabled_log_types" {
   type        = "list"
 }
 
+variable "cluster_log_retention_in_days" {
+  default     = "0"
```
Comment (max-rocket-internet) [inline variables.tf]: I think maybe a default of `30` or `90` might be more sensible. What do you think?

Comment (max-rocket-internet) [reply variables.tf]: Cloudwatch logs can get expensive and on a busy cluster there is a lot of messages. So maybe let's go with `90`.

Comment (yuriipolishchuk) [reply variables.tf]: @max-rocket-internet makes sense

Comment (yuriipolishchuk) [reply variables.tf]: @max-rocket-internet set default retention to 90 days

**Pull request merged**